discord/channel.py

```diff
@@ -169,25 +173,22 @@ class TextChannel(discord.abc.Messageable, discord.abc.GuildChannel, Hashable):
         "default_auto_archive_duration",
     )
 
-    def __init__(self, *, state: ConnectionState, guild: Guild, data: TextChannelPayload):
+    def __init__(self, *, state: ConnectionState, guild: Guild, data: Union[TextChannelPayload, ForumChannelPayload]):
         self._state: ConnectionState = state
         self.id: int = int(data["id"])
         self._type: int = data["type"]
         self._update(guild, data)
 
+    @property
+    def _repr_attrs(self) -> Tuple[str, ...]:
+        return "id", "name", "position", "nsfw", "category_id"
+
     def __repr__(self) -> str:
-        attrs = [
-            ("id", self.id),
-            ("name", self.name),
-            ("position", self.position),
-            ("nsfw", self.nsfw),
-            ("news", self.is_news()),
-            ("category_id", self.category_id),
-        ]
+        attrs = [(val, getattr(self, val)) for val in self._repr_attrs]
         joined = " ".join("%s=%r" % t for t in attrs)
         return f"<{self.__class__.__name__} {joined}>"
 
-    def _update(self, guild: Guild, data: TextChannelPayload) -> None:
+    def _update(self, guild: Guild, data: TextChannelPayload | ForumChannelPayload) -> None:
```

Comment:
iirc `|` for union types is only available in 3.10

```suggestion
    def _update(self, guild: Guild, data: Union[TextChannelPayload, ForumChannelPayload]) -> None:
```